src/shared/Microsoft.Git.CredentialManager/Application.cs

```diff
@@ -105,5 +126,104 @@ protected bool WriteException(Exception ex)
 
             return true;
         }
+
+        #region IConfigurableComponent
+
+        string IConfigurableComponent.Name => "Git Credential Manager";
+
+        Task IConfigurableComponent.ConfigureAsync(
+            IEnvironment environment, EnvironmentVariableTarget environmentTarget,
+            IGitConfiguration configuration, GitConfigurationLevel configurationLevel)
+        {
+            // NOTE: We currently only update the PATH in Windows installations and leave putting the GCM executable
+            //       on the PATH on other platform to their installers.
+            if (PlatformUtils.IsWindows())
+            {
+                string directoryPath = Path.GetDirectoryName(_appPath);
+                if (!environment.IsDirectoryOnPath(directoryPath))
+                {
+                    Context.Trace.WriteLine("Updating environment...");
+                    environment.AddDirectoryToPath(directoryPath, environmentTarget);
+                }
+                else
+                {
+                    Context.Trace.WriteLine("Environment is already set correctly.");
+                }
+            }
+
+            string helperKey = $"{Constants.GitConfiguration.Credential.SectionName}.{Constants.GitConfiguration.Credential.Helper}";
+            string gitConfigAppName = GetGitConfigAppName();
+
+            using (IGitConfiguration targetConfig = configuration.GetFilteredConfiguration(configurationLevel))
+            {
+                string[] currentValues = targetConfig.GetMultivarValue(helperKey, Constants.RegexPatterns.Any).ToArray();
+                if (currentValues.Length != 2 || !string.IsNullOrWhiteSpace(currentValues[0]) ||
```

Comment:
Why are looking for exactly 2 vars?  Why do we only check the first for NullOrWhiteSpace?

Reply:
We require the Git configuration to look like this:

```ini
[credential]
    # technically we don't care how many helper entries appear before
    helper = # empty string here clears any helpers included previously
    helper = manager-core # GCM Core!
    # there must be no more helper entries after
```

I guess really we should be looking for _at least_ 2 vars and then only check the last two for empty & GCM Core. I'll look at changing it.